NOTES.UNIX

```diff
@@ -8,35 +8,53 @@
+ library search path when linking a shared library or an application.
+ They are specified with different linking options depending on operating
+ system and versions thereof, and are talked about differently in their
+ respective documentation, where variations of RPATH are the most usual
+ (to be noted: GNU ld.so systems have two such tags, more on that below).
+
+ Possible options to set the runtime shared library search path are:
+
+    -rpath /whatever/path
+    -R /whatever/path
+    -Wl,-rpath,/whatever/path
+
+ OpenSSL's configuration scripts recognise all these options and pass
+ them to the Makefile that they build.
+ (as a matter of fact, it recognises anything starting with '-Wl,' as a
+ linker option, so for example, HP-UX' '-Wl,+b,/whatever/path' would
+ be used correctly)
+
+ Please do not use verbatim directories in your runtime shared library
+ search path!  Some OpenSSL config targets add an extra directory level
+ for multilib installations.  To help with that, the produced Makefile
+ includes the variable LIBRPATH, which is a convenience variable to be
+ used with the runtime shared library search path options, as shown in
+ this example:
 
     $ ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl \
-        -Wl,-rpath,/usr/local/ssl/lib
+        '-rpath $(LIBRPATH)'
 
- Because the actual library location may vary further (for example on
- multilib installations), there is a convenience variable in Makefile
- that holds the exact installation directory and that can be used like
- this:
+ (the quotes are needed for two reasons: 1) the configuration scripts
+ will otherwise -rpath and $(LIBRPATH) as two entirely separate
+ arguments, and 2) the shell will try to expand $(LIBRPATH))
 
-    $ ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl \
-        -Wl,-rpath,'$(LIBRPATH)'
-
- On modern systems using GNU ld.so, a better choice may be to use the
- new dtags, like this:
+ On modern systems using GNU ld.so, there are two runtime search paths
+ tags to consider, DT_RPATH and DT_RUNPATH.  DT_RUNPATH is considered
+ after the environment variable LD_LIBRARY_PATH, while DT_RPATH is
+ considered before that environment variable (which means that the
+ values in that environment variable won't matter if the library is
+ found in the paths given by DT_RPATH).  Giving the configurations
+ scripts the option '-Wl,-rpath,$(LIBRPATH)' will set DT_RPATH, which
+ may be undesirable.  To set DT_RUNPATH, use the new dtags, like this:
 
     $ ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl \
-        -Wl,--enable-new-dtags,-rpath,'$(LIBRPATH)'
-
- This sets DT_RUNPATH instead of DT_RPATH.  DT_RUNPATH is considered after
- the environment variable LD_LIBRARY_PATH, while DT_RPATH is considered
- before that environment variable (which means that the values in that
- environment variable won't matter if the library is found in the
- paths given by DT_RPATH).
+        '-Wl,--enable-new-dtags,-rpath,$(LIBRPATH)'
```

Comment:
While RPATH/RUNPATH is not really a GNU-ld.so-specific thing, --enable-new-dtags is not universal option. But it's not Linux-specific either, as BSD has it too...

Reply:
Do the BSDs use GNU ld these days?

Reply:
Judging from https://docs.oracle.com/cd/E26502_01/html/E26507/chapter6-42444.html , it seems Solaris has more or less deprecated DT_RPATH.

Reply:
I thought ```--enable-new-dtags``` was approaching universal, as versions age out.

FreeBSD is in the process of switching from GNU ld to LLVM's lld, but I think the other BSDs are still on GNU ld.

Reply:
"not universal" was reference to **all** ELF-based systems, and not Linux distros and BSD versions alone. I mean originally it gave certain impression about RPATH/RUNPATH and --enable-new-dtag availability, which was skewed toward Linux (and BSD).